src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -167,5 +168,32 @@ public ProcessWindowStyle WindowStyle
                 _windowStyle = value;
             }
         }
+
+        internal string BuildArguments()
+        {
+            var stringBuilder = new StringBuilder();
+            AppendArguments(stringBuilder);
+            return stringBuilder.ToString();
+        }
+
+        internal void AppendArguments(StringBuilder stringBuilder)
```

Comment:
Nit: my brain keeps interpreting this as a method that appends arguments to either the `ArgumentList` or `Arguments` properties. I'd probably refactor this as a static method (for example change the original `AppendArguments` method to accept a PSI argument).

Reply:
I was hoping to get the opposite effect: encapsulate of what `ProcessStartInfo` believes is good to append to the provided `StringBuilder` as "arguments". Does it make sense?

Reply:
I think it does, just pointing out that I'm conditioned to think of `Append*` methods as mutating `this` rather than any method arguments. I'd probably name this `AppendArgumentsTo` or `WriteArguments`, but I suppose it doesn't matter that much since this is internals :-)

Reply:
Ok, I am going to rename it to `AppendArgumentsTo`